Code does not compile in g-w after latest PR

### What is wrong?

I update global-workflow to include https://github.com/NOAA-EMC/gfs-utils/pull/105 however I get a build error 

see: /scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/logs/save.gfs_utils.log 

```
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(611): error #5082: Syntax error, found '::' when expecting one of: ) ( , : <END-OF-STATEMENT> ; % [ . = / =>
       intiger(4) :: idx
------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(779): error #5082: Syntax error, found IDENTIFIER 'VFILL' when expecting one of: :: ) ( , : * <END-OF-STATEMENT> ; . (/ + - [ ] /) . ' ** / // ...
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
----------------------------------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(611): error #6404: This name does not have a type, and must have an explicit type.   [INTIGER]
       intiger(4) :: idx
-------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(777): error #6360: A scalar-valued argument is required in this context.   [TRIM]
            idx = findloc(trim(gcf(:)%name_gb2) == 'NSWRF', .true.)
-------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(777): error #6361: An array-valued argument is required in this context.   [FINDLOC]
            idx = findloc(trim(gcf(:)%name_gb2) == 'NSWRF', .true.)
------------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(779): error #6535: This variable or component must be of a derived or structure type.   [FIELD]
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
----------------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(779): error #6460: This is not a component name that is defined in the encompassing structure.   [NE]
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
-------------------------------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(779): error #6158: The structure-name is invalid or is missing.   [FIELD]
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
----------------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(779): error #6385: The highest data type rank permitted is INTEGER(KIND=8).
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
----------------------------------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(779): error #6341: A logical data type is required in this context.
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
----------------------------------------------^
/scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90(611): error #6404: This name does not have a type, and must have an explicit type.   [IDX]
       intiger(4) :: idx
---------------------^
compilation aborted for /scratch1/NCEPDEV/climate/Jessica.Meixner/updateformom6output/global-workflow/sorc/gfs_utils.fd/src/ocnicepost.fd/utils_mod.F90 (code 1)
make[2]: *** [src/ocnicepost.fd/CMakeFiles/ocnicepost.x.dir/build.make:101: src/ocnicepost.fd/CMakeFiles/ocnicepost.x.dir/utils_mod.F90.o] Error 1
```


### What should have happened?

g-w should build pointing to the latest of gfs-utils. 

### What machines are impacted?

All or N/A

### Steps to reproduce

Clone the global-workflow develop,  Update the sorc/gfs-utils to use the top of develop of gfs-utils, and build the g-w.   

### Additional information

Curious how PR https://github.com/NOAA-EMC/gfs-utils/pull/105 was tested and confirmed to produce the correct output to address issue https://github.com/NOAA-EMC/gfs-utils/issues/103 

Pinging: @AminIlia-NOAA @jiandewang @GeorgeGayno-NOAA 

### Do you have a proposed solution?

Fix typo: 

src/ocnicepost.fd/utils_mod.F90 line 611 intiger -> integer   

Add a missing period on line 779

```
-               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
+               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne. vfill ) field(:, n) = field(:, n) + field(:, idx)
```

There might be additional issues after this is fixed though.  I will test this now.

--- a/src/ocnicepost.fd/ocnicepost.F90
+++ b/src/ocnicepost.fd/ocnicepost.F90
@@ -257,39 +257,6 @@ program ocnicepost
           sqrt(rgb2d(:,idx2)**2 + rgb2d(:,idx3)**2)
   end if
 
-!--------------------------------------------------------
-! Write the grib2 files
-!--------------------------------------------------------
-
-  if(write_grib2) then
-   if (allocated(rgb2d) .and. allocated(rgc2d)) then
-      allocate(grib2d(nxr*nyr,nconsd2d+nbilin2d), source=0.0)
-      allocate(g2d(1:nconsd2d+nbilin2d)) 
-      grib2d(:, 1:nbilin2d) = rgb2d
-      grib2d(:, nbilin2d+1:nconsd2d+nbilin2d) = rgc2d
-      g2d(1:nbilin2d) = b2d
-      g2d(nbilin2d+1:nconsd2d+nbilin2d) = c2d
-   else if (allocated(rgb2d)) then
-      allocate(grib2d(nxr*nyr,nconsd2d), source=0.0)
-      allocate(g2d(1:nconsd2d))
-      grib2d(:, 1:nconsd2d) = rgc2d
-      g2d(1:nconsd2d) = c2d
-   else if (allocated(rgc2d)) then
-      allocate(grib2d(nxr*nyr,nbilin2d), source=0.0)
-      allocate(g2d(1:nbilin2d))
-      grib2d(:, 1:nbilin2d) = rgb2d
-      g2d(1:nbilin2d) = b2d
-   end if
-
-   gout = trim(ftype)//'.'//trim(fdst)//'.grib2'
-   if (debug) write(logunit, '(a)')'GRIB2 output file: '//trim(gout)
-   call write_grib2_2d(gout, g2d, (/nxr,nyr/), nconsd2d+nbilin2d, grib2d, vfill)
-
-   if (allocated(rgb3d)) then
-      call write_grib2_3d(gout, b3d, (/nxr,nyr,nlevs/), nbilin3d, rgb3d, vfill)
-   end if
-  end if
-
 
   ! --------------------------------------------------------
   ! write the mapped fields
@@ -414,6 +381,39 @@ program ocnicepost
   write(logunit,'(a)')trim(fout)//' done'
 
  end if
+
+!--------------------------------------------------------
+! Write the grib2 files
+!--------------------------------------------------------
+
+  if(write_grib2) then
+   if (allocated(rgb2d) .and. allocated(rgc2d)) then
+      allocate(grib2d(nxr*nyr,nconsd2d+nbilin2d), source=0.0)
+      allocate(g2d(1:nconsd2d+nbilin2d)) 
+      grib2d(:, 1:nbilin2d) = rgb2d
+      grib2d(:, nbilin2d+1:nconsd2d+nbilin2d) = rgc2d
+      g2d(1:nbilin2d) = b2d
+      g2d(nbilin2d+1:nconsd2d+nbilin2d) = c2d
+   else if (allocated(rgb2d)) then
+      allocate(grib2d(nxr*nyr,nconsd2d), source=0.0)
+      allocate(g2d(1:nconsd2d))
+      grib2d(:, 1:nconsd2d) = rgc2d
+      g2d(1:nconsd2d) = c2d
+   else if (allocated(rgc2d)) then
+      allocate(grib2d(nxr*nyr,nbilin2d), source=0.0)
+      allocate(g2d(1:nbilin2d))
+      grib2d(:, 1:nbilin2d) = rgb2d
+      g2d(1:nbilin2d) = b2d
+   end if
+
+   gout = trim(ftype)//'.'//trim(fdst)//'.grib2'
+   if (debug) write(logunit, '(a)')'GRIB2 output file: '//trim(gout)
+   call write_grib2_2d(gout, g2d, (/nxr,nyr/), nconsd2d+nbilin2d, grib2d, vfill)
+
+   if (allocated(rgb3d)) then
+      call write_grib2_3d(gout, b3d, (/nxr,nyr,nlevs/), nbilin3d, rgb3d, vfill)
+   end if
+  end if
 
  stop
 

--- a/src/ocnicepost.fd/utils_mod.F90
+++ b/src/ocnicepost.fd/utils_mod.F90
@@ -617,7 +617,7 @@ contains
        integer(4) :: jgdt(19), jpdt(15), idrtmpl(16)
        integer(4) :: igds(5)
        integer(4) :: numcoord, ibmap
-       real(4)    :: coordlist
+       real(8)    :: coordlist
        integer(4) :: n, lon0, lon1, lat0, lat1
        integer(4) :: ideflist, idefnum
        logical*1 :: bmp(dims(1)*dims(2)) 
@@ -724,14 +724,6 @@ contains
             return
          end if
 
-       ! Compute max, min, and mean
-         max_val = maxval(field(:,n), mask = field(:,n) .ne. vfill)
-         min_val = minval(field(:,n), mask = field(:,n) .ne. vfill)
-         mean_val = sum(field(:,n), mask = field(:,n) .ne. vfill) / count(field(:,n) .ne. vfill)
-
-         if (debug) then
-            write(logunit, *) 'Variable_name, max, min, mean: ', gcf(n)%var_name, max_val, min_val, mean_val
-         end if
 
          call addgrid(cgrib, max_bytes, igds, jgdt, igdtlen, ideflist, idefnum, ierr) 
          if (ierr /= 0) then
@@ -767,21 +759,25 @@ contains
          coordlist=0.  ! needed for hybrid vertical coordinate
 
          ibmap = 0     ! Bitmap indicator ( see Code Table 6.0 ) 
-         bmp=.true.
+         bmp=.false.
 
          if ((trim(gcf(n)%name_gb2) .eq. 'WTMP' ) .or.  (trim(gcf(n)%name_gb2) .eq. 'ICETMP' )) then 
             where ( field(:,n) .ne. vfill ) field(:,n) = field(:,n) + 273.15
          endif
 
          if (trim(gcf(n)%name_gb2) == 'THFLX') then
+           idx=-1      
            do i = 1, size(gcf)
              if (trim(gcf(i)%name_gb2) .eq. 'NSWRF') then
-                idx = i
-                 exit
+              idx = i
+              exit
              end if
            end do
            if (idx > 0) then
-               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne. vfill ) field(:, n) = field(:, n) + field(:, idx)
+              where ( (field(:, n) .ne. vfill) .and. (field(:, idx) .ne. vfill) ) 
+                  field(:, n) = field(:, n) + field(:, idx)
+                  bmp(:)=.true.
+              end where
            else
               print *, "Fatal error: NSWRF must be in parameter list"
               stop 99
@@ -789,7 +785,7 @@ contains
          end if
 
 
-         where ( field(:,n) .eq. vfill )  bmp(:)= .false.
+         where ( field(:,n) .ne. vfill )  bmp(:)= .true.
 
          !  Create Section 5 parametrs   
          idrtnum = 0                            ! Template 5.0 (Grid Point Data - simple Packing)
@@ -812,8 +808,14 @@ contains
          tmpfld=0 
          tmpfld=real(field(:,n), 8)
 
+         ! Compute max, min, and mean
+         max_val = maxval(field(:,n), mask = field(:,n) .ne. vfill)
+         min_val = minval(field(:,n), mask = field(:,n) .ne. vfill)
+         mean_val = sum(field(:,n), mask = field(:,n) .ne. vfill) / count(field(:,n) .ne. vfill)
 
-         if (debug) write(logunit, *) 'Variable_name, max, min, mean, count: ', gcf(n)%var_name, max_val, min_val, mean_val, count_val
+         if (debug) then
+            write(logunit, *) 'Variable_name, max, min, mean: ', gcf(n)%var_name, max_val, min_val, mean_val
+         end if
 
          call addfield(cgrib, max_bytes, ipdtnum, jpdt, ipdtlen, coordlist, numcoord, &
          idrtnum, idrtmpl, idrtlen, tmpfld, npt, ibmap, bmp, ierr)
@@ -871,7 +873,7 @@ contains
    integer(4) :: jgdt(19), jpdt(15), idrtmpl(16)
    integer(4) :: igds(5)
    integer(4) :: numcoord, ibmap
-   real(4):: coordlist
+   real(8):: coordlist
    integer(4) :: ideflist, idefnum
    logical*1 :: bmp( dims(1) * dims(2) ) 
 

--- a/src/ocnicepost.fd/utils_mod.F90
+++ b/src/ocnicepost.fd/utils_mod.F90
@@ -608,7 +608,7 @@ contains
        integer(4) :: fortime, dij, npt
        CHARACTER(len=1),allocatable,dimension(:) :: cgrib
        real(8) :: tmpfld(size(field,1))
-       intiger(4) :: idx
+       integer(4) :: idx, i
 
        ! GRIB2 metadata arrays
        integer(4) :: listsec0(2), listsec1(13)
@@ -774,13 +774,18 @@ contains
          endif
 
          if (trim(gcf(n)%name_gb2) == 'THFLX') then
-            idx = findloc(trim(gcf(:)%name_gb2) == 'NSWRF', .true.)
-            if (idx > 0) then
-               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)
-            else
+           do i = 1, size(gcf)
+             if (trim(gcf(i)%name_gb2) .eq. 'NSWRF') then
+                idx = i
+                 exit
+             end if
+           end do
+           if (idx > 0) then
+               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne. vfill ) field(:, n) = field(:, n) + field(:, idx)
+           else
               print *, "Fatal error: NSWRF must be in parameter list"
               stop 99
-            end if
+           end if
          end if
 
 
@@ -808,7 +813,7 @@ contains
          tmpfld=real(field(:,n), 8)
 
 
-         write(logunit, *) 'Variable_name, max, min, mean, count: ', gcf(n)%var_name, max_val, min_val, mean_val, count_val
+         if (debug) write(logunit, *) 'Variable_name, max, min, mean, count: ', gcf(n)%var_name, max_val, min_val, mean_val, count_val
 
          call addfield(cgrib, max_bytes, ipdtnum, jpdt, ipdtlen, coordlist, numcoord, &
          idrtnum, idrtmpl, idrtlen, tmpfld, npt, ibmap, bmp, ierr)

--- a/src/ocnicepost.fd/utils_mod.F90
+++ b/src/ocnicepost.fd/utils_mod.F90
@@ -812,7 +812,8 @@ contains
          mean_val = sum(field(:,n), mask = field(:,n) .ne. vfill) / count(field(:,n) .ne. vfill)
 
          if (debug) then
-            write(logunit, *) 'Variable_name, max, min, mean: ', gcf(n)%var_name, max_val, min_val, mean_val
+             write(logunit, '(a, 1x, a, 1x, f10.4, 1x, f10.4, 1x, f10.4)') &
+              'Variable_name, max, min, mean: ', gcf(n)%var_name, max_val, min_val, mean_val
          end if
 
          call addfield(cgrib, max_bytes, ipdtnum, jpdt, ipdtlen, coordlist, numcoord, &

--- a/src/ocnicepost.fd/utils_mod.F90
+++ b/src/ocnicepost.fd/utils_mod.F90
@@ -778,7 +778,7 @@ contains
                   bmp(:)=.true.
               end where
            else
-              print *, "Fatal error: NSWRF must be in parameter list"
+              write(0,'(a)')'FATAL ERROR: NSWRF must be in parameter list'
               stop 99
            end if
          end if

